engine/src/main/java/org/terasology/math/ChunkMath.java

```diff
@@ -64,35 +67,188 @@ public static int calcChunkPosZ(int z, int chunkPowerZ) {
+        return calcChunkPos(Math.roundHalfUp(pos.x()), Math.roundHalfUp(pos.y()), Math.roundHalfUp(pos.z()), ChunkConstants.POWER_X, ChunkConstants.POWER_Y, ChunkConstants.POWER_Z, dest);
+    }
+
+    /**
+     * The position relative to the size of chunk with the given chunk power
+     *
+     * <p>Chunk size is in powers of 2 (2, 4, 8, 16, ...)</p>
+     *
+     * @param pos absolute position of the block
+     * @param chunkPower the size of the chunk in powers 2
+     * @param dest will hold the result
+     * @return dest
+     */
+    public static org.joml.Vector3i calcChunkPos(Vector3fc pos, Vector3ic chunkPower, org.joml.Vector3i dest) {
+        return calcChunkPos(Math.roundHalfUp(pos.x()), Math.roundHalfUp(pos.y()), Math.roundHalfUp(pos.z()),
+            chunkPower, dest);
+    }
+
+    /**
+     * The position relative to the size of chunk with the given power ({@link ChunkConstants#POWER_X}, {@link ChunkConstants#POWER_Y}, {@link ChunkConstants#POWER_Z})
+     *
+     * <p>Chunk size is in powers of 2 (2, 4, 8, 16, ...)</p>
+     *
+     * @param pos absolute position of the block
+     * @param dest will hold the result
+     * @return dest
+     */
+    public static org.joml.Vector3i calcChunkPos(Vector3ic pos, org.joml.Vector3i dest) {
+        return calcChunkPos(pos.x(), pos.y(), pos.z(), ChunkConstants.POWER_X, ChunkConstants.POWER_Y, ChunkConstants.POWER_Z, dest);
+    }
+
+    /**
+     * The position relative to the size of chunk with the given power ({@link ChunkConstants#POWER_X}, {@link ChunkConstants#POWER_Y}, {@link ChunkConstants#POWER_Z})
+     *
+     * <p>Chunk size is in powers of 2 (2, 4, 8, 16, ...)</p>
+     *
+     * @param x absolute x component of the block
+     * @param y absolute y component of the block
+     * @param z absolute z component of the block
+     * @param dest will hold the result
+     * @return dest
+     */
+    public static org.joml.Vector3i calcChunkPos(int x, int y, int z, org.joml.Vector3i dest) {
+        return calcChunkPos(x, y, z, ChunkConstants.POWER_X, ChunkConstants.POWER_Y, ChunkConstants.POWER_Z, dest);
+    }
+
+    /**
+     * The position relative to the size of chunk with the given chunk power
+     *
+     * <p>Chunk size is in powers of 2 (2, 4, 8, 16, ...)</p>
+     *
+     * @param pos the absolute position of the block
+     * @param chunkPower the size of the chunk in powers of 2
+     * @param dest will hold the result
+     * @return dest
+     */
+    public static org.joml.Vector3i calcChunkPos(Vector3ic pos, Vector3ic chunkPower, org.joml.Vector3i dest) {
+        return calcChunkPos(pos.x(), pos.y(), pos.z(), chunkPower.x(), chunkPower.y(), chunkPower.z(), dest);
+    }
+
+    /**
+     * The position relative to the size of chunk with the given chunk power
+     *
+     * <p>Chunk size is in powers of 2 (2, 4, 8, 16, ...)</p>
+     *
+     * @param x absolute x component of the block
```

Comment:
Is "component" a math term here? To me as a mundane develop it makes me think about our ES Components. If not a needed math term could it be named differently?

Reply:
I just follow what JOML is doing with referring to vector components. do you have a better name we can use? 

![image](https://user-images.githubusercontent.com/854359/83709860-40617500-a5d4-11ea-9d69-fc11f475a978.png)

Reply:
x y and z _coordinates_ ?

Reply:
Mathemtically, it's more common to talk about "the components of a vector" when interpreting that vector as merely an n-tuple of numbers. It seems that when the components are interpreted as a position in some kind of coordinate system (like cartesian) then it's more common to talk about "coordinates".

Reply:
yeah, component was the standard term for this in my math/physics education. unfortunate terminology collision in an environment that works with entity-component-system so much.